src/main/java/com/revature/services/QuestionService.java

```diff
@@ -1,20 +1,124 @@
 package com.revature.services;
 
+import java.util.Optional;
+
 import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.data.domain.Page;
 import org.springframework.data.domain.Pageable;
+import org.springframework.http.HttpStatus;
 import org.springframework.stereotype.Service;
+import org.springframework.web.client.HttpClientErrorException;
 
+import com.revature.entities.Answer;
 import com.revature.entities.Question;
+import com.revature.entities.User;
+import com.revature.DTOs.RSSAccountDTO;
+import com.revature.repositories.AnswerRepository;
 import com.revature.repositories.QuestionRepository;
 
 @Service
 public class QuestionService {
 
+	@Autowired 
+	RSSService rssService;	
+	
 	@Autowired
 	QuestionRepository questionRepository;
 	
+	@Autowired
+	AnswerRepository answerRepository;
+	
+	public QuestionService(QuestionRepository questionRepository) {
+		this.questionRepository = questionRepository;
+	}
+	
+	/**@author ken*/
 	public Page<Question> getAllQuestions(Pageable pageable){
 		return questionRepository.findAll(pageable);
 	}
+
+	/**@author ken*/
+	public Page<Question> getAllQuestionsByUserId(Pageable pageable, int id){
+		return questionRepository.getAllQuestionsByUserID(pageable, id);		
+	}
+	
+	/** @Author James Walls */
+	/** Adds new questions and updates existing ones. */
+	public Question save(Question question) {
+		return questionRepository.save(question);
+	}
+
+	/** 
+	 * @author Hugh Thornhill 
+	 * @return This updates the AcceptedAnswerId on the Questions table.
+	 * This indicates that the user who posted the question has accepted an
+	 * answer as the best one. The comments below indicate how the rest of the
+	 * implementation can be written out.
+	 */
+	public Question updateQuestionAcceptedAnswerId(Question question) {
+		if(question.getId() == 0) {
+			throw new HttpClientErrorException(HttpStatus.BAD_REQUEST);
+		}
+		
```

Comment:
Should there be logic here to make sure question is valid?

Reply:
I didn't have time to implement the logic to make sure the request was valid. The comments within the method list the general way it can be written out. I also made an entry about this in the wiki under the Bugs page.

Reply:
Hugh - is the need for this to be completed documented anywhere?  If not, you can create an issue notating the need for the validation logic to be completed.

Reply:
Yes it is documented under the Bugs section of the Wiki. I'll make an issue for it too.